gadget/validate.go

```diff
@@ -112,6 +112,9 @@ func ruleValidateVolumes(vols map[string]*Volume, model Model, extra *Validation
 		if err := ruleValidateVolume(v, hasModes); err != nil {
 			return fmt.Errorf("invalid volume %q: %v", name, err)
 		}
+		if v.Schema == schemaEMMC && len(vols) > 2 {
```

Comment:
btw should we check  that we have at most one volume with emmc schema?

Reply:
@bboozzoo I think we should avoid that with the coming changes for volume-assigments here, as we might have multiple eMMC schemas for gadget reuse. Instead this should be checked during runtime. I'll add a task for this under the epic so we remember this.